src/GitTfs.VsCommon/TfsHelper.Vs2017Base.cs

```diff
@@ -61,7 +72,51 @@ protected override bool HasWorkItems(Changeset changeset)
         /// </returns>
         protected string GetVsInstallDir()
         {
-            return @"C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise";
+            if (myVisualStudioInstallationPath != null)
+                return myVisualStudioInstallationPath;
+
+            var setupConfiguration = (ISetupConfiguration2)GetSetupConfiguration();
+            IEnumSetupInstances instancesEnumerator = setupConfiguration.EnumAllInstances();
+
+            int fetched;
+            var instances = new ISetupInstance[1];
+            while (true)
+            {
+                instancesEnumerator.Next(1, instances, out fetched);
+                if (fetched <= 0)
+                    break;
+
+                var instance = (ISetupInstance2)instances[0];
+                if (!Version.TryParse(instance.GetInstallationVersion(), out Version version))
+                {
+                    Trace.TraceError("Failed to retrieve version. Skipping VS instance.");
+                    continue;
+                }
+
+                if (version.Major != myMajorVersion)
+                {
+                    continue;
+                }
+
+                if (myVisualStudioInstallationPath != null)
+                {
+                    Trace.TraceWarning("Already found a Visual Studio version. Ignoring version at {0}", instance.GetInstallationPath());
+                    continue;
+                }
```

Comment:
Why is it still iterating at this point?

Reply:
I fought with myself here when implementing this, but then decidded to continue to iterate so we will at least show
**all** the  editions & versions of one VS major, e.g. all VS2017 instances on the machine.
Because at least I didn't find any garantuees if the order of the returned intances is always deterministic, e.g. could be that one git-tfs call will find VS2017 Enterprise whereas the next run will find the VS2017 Professional which is also installed, which is problematic in case of (yet to come) checkin policies, because there can be different policies installed in the different versions.

To sum it up, in case of multiple versions with the same major, I wanted to give the user a chance to see which one
is taken and which other versions of the same major are potentially there but are not used, as I explicitley stated in the commit messages that supporting multiple versions of the same major is out of scope right now.

Reply:
> To sum it up, in case of multiple versions with the same major, I wanted to give the user a chance to see which one is taken and which other versions of the same major are potentially there but are not used, as I explicitley stated in the commit messages that supporting multiple versions of the same major is out of scope right now.

This seems like a pretty rare edge case, where there are multiple versions and one of them won't work and git-tfs chooses the one that won't work. What do you think about simplifying this so that it stops looping after it finds a working installation? If it does end up being a problem, this can be changed later to let the user choose an install.

Alternatively, maybe reduce the log level to 'Info' so that it's not as noisy in benign cases.

Reply:
For checkin policies it is not a rare case, because at least here it happened to me that I simply had forgotten to install the checkin policies in both VS2017 versions and due to bad luck git-tfs has chosen to use the one which didn't have it installed. And in that case it was quite confusing why it didn't work.
But I agree with you that the rest of the functionality shouldn't care which VS2017 edition (or minor) is used. 

So the additional looping (and therefore the logging) only has an effect if more than one version of the **same** studio major is installed.  I think that having mutliple VS2017 instances installed is already a rare case , but I wanted to provide at least some breadcrumbs in case something goes wrong,  as it is otherwise not easy to figure out.

All others which have e.g. only one VS2019 and one VS2017 installed should not see any additional logging output, as that case is alread covered  with this comparision in the loop
```csharp
           if (version.Major != myMajorVersion)
           {
                    continue;
           }
```
At least that was my intention when writing the code.

Reply:
👍 I'll defer to you on this.

Reply:
Marked as resolved as I think it has value to show all versions, as explained above